component/common/src/main/java/org/exoplatform/platform/common/rest/services/WhoIsOnLinePortlet/WhoIsOnLineRestService.java

```
@@ -0,0 +1,140 @@
+                JSONObject json = new JSONObject();
+                Identity userIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, user,false);
+                Profile userProfile = userIdentity.getProfile();
+                json.put("id", userProfile.getId());
+                json.put("profileUrl", LinkProvider.getUserProfileUri(userIdentity.getRemoteId()));
+                json.put("avatar", LinkProviderUtils.getUserAvatarUrl(userProfile));
+                jsonArray.put(json);
+            }
+            return jsonArray;
+
+        } catch (Exception e) {
+            return jsonArray;
+        }
+    }
+
+    private JSONArray getOnlineUsers(String userId) {
+        JSONArray jsonArray = new JSONArray();
+        try {
+            List<UserStateModel> users = userStateService.online();
+            Collections.reverse(users);
+            if (users.size() > MAX_USER) {
+                users = users.subList(0, INDEX_USER);
+            }
+
+            String superUserName = System.getProperty("exo.super.user");
+            for (UserStateModel userModel : users) {
+                String user = userModel.getUserId();
+                if (user.equals(userId) || user.equals(superUserName) || INVISIBLE.equals(userModel.getStatus())) continue;
+                JSONObject json = new JSONObject();
+                Identity userIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, user,false);
+                Profile userProfile = userIdentity.getProfile();
+                json.put("id", userProfile.getId());
+                json.put("profileUrl", LinkProvider.getUserProfileUri(userIdentity.getRemoteId()));
+                json.put("avatar", LinkProviderUtils.getUserAvatarUrl(userProfile));
+                jsonArray.put(json);
+            }
+            return jsonArray;
+
+        } catch (Exception e) {
+            return jsonArray;
```

Same remark about the swallowed exception.